tests/test_zing_stats.py

```diff
@@ -51,16 +51,89 @@ def test_parse_gerrit_change_message():
+                "gravatar_id": "",
+                "html_url": "https://github.example.com/a_user",
+                "id": 19638,
+                "login": "a_user",
+                "organizations_url": "https://github.example.com/api/v3/users/a_user/orgs",
+                "received_events_url": "https://github.example.com/api/v3/users/a_user/received_events",
+                "repos_url": "https://github.example.com/api/v3/users/a_user/repos",
+                "site_admin": "false",
+                "starred_url": "https://github.example.com/api/v3/users/a_user/starred{/owner}{/repo}",
+                "subscriptions_url": "https://github.example.com/api/v3/users/a_user/subscriptions",
+                "type": "User",
+                "url": "https://github.example.com/api/v3/users/a_user"
+            }
+        },
+        {
+            "body": "Build succeeded\n\n- http://logs.example.net/check-github/foo/api/111153/151255557209.72/foo-example-check : SUCCESS in 2m 38s\n- http://logs.example.net/check-github/foo/api/111153/151112557209.72/foo-sec-scan : SUCCESS in 4s (non-voting)\n- http://logs.example.net/check-github/foo/api/122153/151332557209.72/another-scan : SUCCESS in 4s (non-voting)\n",
+            "created_at": "2017-12-06T10:49:06Z",
+            "html_url": "https://github.example.com/foo/api/pull/1153#issuecomment-429779",
+            "id": 429779,
+            "issue_url": "https://github.example.com/api/v3/repos/foo/api/issues/1153",
+            "updated_at": "2017-12-06T10:49:06Z",
+            "url": "https://github.example.com/api/v3/repos/foo/api/issues/comments/429779",
+            "user": {
+                "avatar_url": "https://avatars.github.example.com/u/19638?",
+                "events_url": "https://github.example.com/api/v3/users/a_user/events{/privacy}",
+                "followers_url": "https://github.example.com/api/v3/users/a_user/followers",
+                "following_url": "https://github.example.com/api/v3/users/a_user/following{/other_user}",
+                "gists_url": "https://github.example.com/api/v3/users/a_user/gists{/gist_id}",
+                "gravatar_id": "",
+                "html_url": "https://github.example.com/a_user",
+                "id": 19638,
+                "login": "a_user",
+                "organizations_url": "https://github.example.com/api/v3/users/a_user/orgs",
+                "received_events_url": "https://github.example.com/api/v3/users/a_user/received_events",
+                "repos_url": "https://github.example.com/api/v3/users/a_user/repos",
+                "site_admin": "false",
+                "starred_url": "https://github.example.com/api/v3/users/a_user/starred{/owner}{/repo}",
+                "subscriptions_url": "https://github.example.com/api/v3/users/a_user/subscriptions",
+                "type": "User",
+                "url": "https://github.example.com/api/v3/users/a_user"
+            }
+        },
+    ]
```

Comment:
future improvement suggestion, move these into a separate yaml file and load at test time. Should be able to create a base test class with a setUp() method that loads message files based on test function name just before the test is run

That way you could create sub classes that use the corresponding `parse_*_message`  as many times as necessary and then assert on the contents.

Will allow building a lot of scenarios easily for different parsing behaviour.

Reply:
Great point - the testing side of the project needs radical improvement and this is good feedback into that work.